solr/solrj/src/java/org/apache/solr/client/solrj/io/stream/MonitorStream.java

```diff
@@ -0,0 +1,235 @@
+    } else {
+      id = ((StreamExpressionValue) idExpression.getParameter()).getValue();
+    }
+
+    StreamExpressionNamedParameter operatorExpression = factory.getNamedOperand(expression, OPERATOR_PARAM);
+    String operator = null;
+    if(operatorExpression == null) {
+      throw new IOException("Invalid expression for operator parameter");
+    } else {
+      operator = ((StreamExpressionValue) operatorExpression.getParameter()).getValue();
+    }
+
+    StreamExpressionNamedParameter flParam = factory.getNamedOperand(expression, FIELD_PARAM);
+    if(null == flParam) {
+      throw new IOException("invalid TopicStream fl cannot be null");
+    }
+
+    ModifiableSolrParams solrParams = new ModifiableSolrParams();
+    for(StreamExpressionNamedParameter namedParam : namedParams){
+      String paramName = namedParam.getName();
+      if(!paramName.equals(ID) && !paramName.equals(OPERATOR_PARAM) && !paramName.equals(URL_PARAM)) {
+        solrParams.set(paramName, namedParam.getParameter().toString().trim());
+      }
+    }
+
+    if (operator.equals(UPDATE_OPERATOR)) {
+      String destinationCollection = factory.getValueOperand(expression, 3);
+      verifyCollectionName(destinationCollection, expression);
+      init(id, destinationCollection, topicCollection, zkHost, solrParams);
+
+    } else {
+      StreamExpressionNamedParameter urlExpression = factory.getNamedOperand(expression, URL_PARAM);
+      String url = null;
+      if(urlExpression == null) {
+        throw new IOException("Invalid expression for url parameter ");
+      } else {
+        url = ((StreamExpressionValue) urlExpression.getParameter()).getValue();
+      }
+      init(id, topicCollection, zkHost, solrParams, url);
+    }
+  }
+
+  public MonitorStream(String id, String topicCollection, String destinationCollection, String zkHost, SolrParams solrParams) throws IOException{
+    init(destinationCollection, topicCollection, id, zkHost, solrParams);
+  }
+
+  public MonitorStream(String id, String topicCollection, String zkHost, SolrParams solrParams, String url) throws IOException{
+    init(topicCollection, id, zkHost, solrParams, url);
+  }
+
+  public void init(String id, String topicCollection, String zkHost, SolrParams solrParams, String url) throws IOException {
+    long initialCheckpoint = -1;
+    long checkpointEvery = -1;
+    long runInterval = 50;
+    int queueSize = 0;
+
+    String topicId = "monitor-".concat(UUID.randomUUID().toString());
+    TopicStream topicStream = new TopicStream(zkHost, topicCollection, topicCollection, topicId,
+      initialCheckpoint, checkpointEvery, solrParams);
+
+    AlertStream alertStream = new AlertStream(topicStream, url);
+    this.daemonStream = new DaemonStream(alertStream, id, runInterval, queueSize);
+  }
+
+  public void init(String id, String destinationCollection, String topicCollection, String zkHost, SolrParams solrParams) throws IOException{
```

Comment:
There looks like a lot of repeated logic in these init methods, might want to have one call the other

Reply:
Also similarly to AlertStream, is there a reason to have these init methods, versus just putting the logic into the second two constructors?

Reply:
The init method follows the same format as the constructors for other Stream expressions. Also if calling one constructor from another have the problem mentioned above. Moved repeated logic to a separate function instead.